### PR TITLE
implement array_shuffle

### PR DESCRIPTION
**Pull request by gushul**
Implemented `array_shuffle` function for postgres under https://github.com/diesel-rs/diesel/issues/4153

### CODE REVIEW

**Comment by weiznich** [comment]
Seems like the CI job fails as we use a Postgres version for our tests that does not support `array_shuffle` yet. I will look at how to solve that next week.

File: diesel/src/pg/expression/functions.rs [hunk 1]
```diff
@@ -1423,3 +1423,30 @@ define_sql_function! {
...
+    /// # }
+    /// #
+    /// # fn run_test() -> QueryResult<()> {
+    /// #     use diesel::dsl::array_shuffle;
+    /// #     use diesel::sql_types::{Array, Integer};
+    /// #     let connection = &mut establish_connection();
+    /// let shuffled = diesel::select(array_shuffle::<Array<Integer>, _>(vec![1, 2, 3, 4, 5]))
+    ///     .get_result::<Vec<i32>>(connection)?;
+    /// assert_eq!(5, shuffled.len());
+    /// assert!(shuffled != vec![1, 2, 3, 4, 5]);  // It's very unlikely to get the same order
```
[the remark this answers is not in the document]
Comment (Mingun) [reply diesel/src/pg/expression/functions.rs:1447]: You can set bits corresponding to numbers in vec and check that the result is 31.

**Comment by weiznich** [comment]
I had another look at this and I think the best way forward for now is to mark this test as `no_run` doc-test as we cannot expect that all contributors have postgres 16 installed locally. That would mean we compile but don't run the test, which is fine for this case. We also should have a comment there saying that the `no_run` setting is there because the test requires postgres 16.

File: diesel/src/pg/expression/functions.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (kw217) [reply diesel/src/pg/expression/functions.rs:1447]: Yep, or sort the result in Rust and compare with the sorted array.

**Review by weiznich — APPROVED** [review]
I've just changed the test to `no_run`, otherwise this looks good to me. Thanks for working on this :+1:

**Pull request merged**